src/main/java/com/shaft/support/DBActions.java

```diff
@@ -0,0 +1,155 @@
+package com.shaft.support;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.testng.Assert;
+
+import com.shaft.io.ReportManager;
+
+public class DBActions {
+
+	Connection connection;
+	Statement statement;
+	ResultSet resultSet;
+	String dbType;
+	String dbServerIP;
+	String dbPort;
+	String dbName;
+	String username;
+	String password;
+
+	public DBActions(String dbType, String dbServerIP, String dbPort, String dbName, String username, String password) {
+		this.dbType = dbType;
+		this.dbServerIP = dbServerIP;
+		this.dbPort = dbPort;
+		this.dbName = dbName;
+		this.username = username;
+		this.password = password;
+	}
+
+	private void passAction(String actionName, String testData, String log) {
+		String message = "Successfully performed action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Passed", "DB Connection Log", log);
+		}
+	}
+
+	private void passAction(String actionName, String testData) {
+		passAction(actionName, testData, null);
+	}
+
+	private void failAction(String actionName, String testData, String log) {
+		String message = "Failed to perform action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 */
+	public void createConnection() {
```

Comment:
it is a must for this method to be private, and to be called automatically at the start of the executeSelectQuery() method. and the javadocs should be updated as both the exceptions are now handled and not thrown

Reply:
Done